tests/conftest.py

```diff
@@ -21,6 +21,22 @@
 from testing.util import cwd
 
 
+@pytest.fixture(autouse=True)
+def no_warnings(recwarn):
+    yield
+    warnings = tuple(
+        warning for warning in recwarn
+        # python2 + pypy warn:
+        # ImportWarning: Not importing directory '...' missing __init__.py
+        if not (
+            isinstance(warning.message, ImportWarning) and
+            str(warning.message).startswith('Not importing directory ') and
+            str(warning.message).endswith(' missing __init__.py')
+        )
```

Comment:
you probably don't need this filtering here (I would hope?) -- I think this was just due to testing a namespace package in the other library

Reply:
Nope, it is needed because on Python 3.6 and 3.7 there are a lot of these warnings.

Reply:
Ah. I just saw the check is not good on Python 6:
```python
python3.6/importlib/_bootstrap_external.py:426: ImportWarning: Not importing directory ~/python3.6/site-packages/aspy: missing __init__
```
There is no `.py`. I will change that.

Reply:
Ah lol of course, the same name space package is still in play

Reply:
Tests are failing on Python 3.7. But the warnings list is not helpful. I updated to patch to have a more human message.